src/Compilers/CSharp/Portable/Symbols/TypeWithState.cs

```diff
@@ -19,13 +18,17 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 
         public static TypeWithState ForType(TypeSymbol type)
         {
-            var state = type?.CanContainNull() != false ? NullableFlowState.MaybeNull : NullableFlowState.NotNull;
-            return new TypeWithState(type, state);
+            return Create(type, NullableFlowState.MaybeDefault);
```

Comment:
>MaybeDefault [](start = 50, length = 12)

Isn't `MaybeDefault` only supposed to be used for unconstrained type parameters?

Reply:
If the type is not an unconstrained type parameter, `TypeWithState.Create()` will use `MaybeNull` instead.

---
In reply to: [347686283](https://github.com/dotnet/roslyn/pull/39778#discussion_r347686283) [](ancestors = 347686283)

Reply:
This is relying on an implementation detail of `Create` though. Same type so not that dangerous. 

Did you consider having the `defaultState` of `Create` be an optional parameter with the default of `MaybeDefault` instead?

---
In reply to: [347730782](https://github.com/dotnet/roslyn/pull/39778#discussion_r347730782) [](ancestors = 347730782,347686283)